core/retry.go

```diff
@@ -0,0 +1,128 @@
+package retry
+
+import (
+	"fmt"
+	"time"
+
+	"github.com/cenkalti/backoff/v4"
+)
+
+/*
+This retry library was inspired by and uses Cenk Alti (https://github.com/cenkalti) backoff library (https://github.com/cenkalti/backoff).
+We would like to thank him for his great work.
+*/
+
+/*
+Note we use a custom Permanent error type for asserting Permanent Erros within the retry library.
+We do not implement an explicit Transient error type and operate under the assumption that all errors that are not Permanent are Transient.
+*/
+type PermanentError struct {
+	Inner error
+}
+
+func (e PermanentError) Error() string { return e.Inner.Error() }
+func (e PermanentError) Unwrap() error {
+	return e.Inner
+}
+func (e PermanentError) Is(err error) bool {
+	_, ok := err.(PermanentError)
+	return ok
+}
+
+const MinDelay = 1000
+const RetryFactor = 2
+const NumRetries = 3
+const MaxInterval = 60000
+const MaxElapsedTime = 0
+
+// Same as Retry only that the functionToRetry can return a value upon correct execution
+func RetryWithData[T any](functionToRetry func() (T, error), minDelay uint64, factor float64, maxTries uint64, maxInterval uint64, maxElapsedTime uint64) (T, error) {
```

Comment:
This params should be `time.Duration` instead of `uint`

Reply:
addressed